File: Client/Frontend/Browser/BraveGetUA.swift
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import Foundation
import WebKit
import Shared
import BraveShared

class BraveGetUA: TabContentScript {
    fileprivate weak var tab: Tab?

    required init(tab: Tab) {
        self.tab = tab
    }

    static func name() -> String {
        return "BraveGetUA"
    }

    func scriptMessageHandlerName() -> String? {
        return BraveGetUA.name()
    }

    func userContentController(_ userContentController: WKUserContentController, didReceiveScriptMessage message: WKScriptMessage) {
        // 🙀 😭 🏃‍♀️💨
    }

    static var isActivated: Bool {
        return true
    }
}

File: ios/brave-ios/Sources/Brave/Frontend/UserContent/UserScripts/AllFrames/AtDocumentStart/BraveGetUA.js
// Copyright (c) 2020 The Brave Authors. All rights reserved.
// This Source Code Form is subject to the terms of the Mozilla Public
// License, v. 2.0. If a copy of the MPL was not distributed with this file,
// You can obtain one at https://mozilla.org/MPL/2.0/.

'use strict'

Object.defineProperty(navigator, 'brave', {
  enumerable: false,
  configurable: true,
  writable: false,
  value: Object.freeze({
    isBrave: () => new Promise(res => res(true))
  })
})
